### PR TITLE
Change Key Generation to take a URI

### LINKED ISSUE #279: Create keys in a specific token/slot
**Describe the feature**

When generating keys, it would be nice to set the specific token/slot where the key needs to be generated.

**Expected behavior**

When generating the key with for example EVP_PKEY_generate(), the key is created in the specified token. The token I expect can be set with in the OSSL_PARAM array, similar to the object/label: 
```
        params[0] = OSSL_PARAM_construct_utf8_string("pkcs11_key_label", label, 0);
```

Alternatively, one param entry could take an RFC 7512 PKCS#11 string that may contain the `object` and `token`.

**Additional context**

For context, I am working on making DNSSEC policy in BIND 9 work with HSMs using the OpenSSL PKCS#11 Provider.

### PR DESCRIPTION
**Pull request by simo5**
Drop passing in individual separate parameters for Label or CKA_ID, rely on URI to provide those (object/id) and also use it to select which token to generate the key on.

Fixes #279

### CODE REVIEW

**Comment by simo5** [comment]
@matje PTAL
@sahanaprasad07 can you review?

**Comment by sahanaprasad07** [comment]
LGTM

**Comment by matje** [comment]
Changes look sane to me, AFAICS. The code also works with my BIND branch after making the `params` modifications. So LGTM.

**Review by sahanaprasad07 — APPROVED** [review]
LGTM

File: src/keymgmt.c [hunk 1]
```diff
@@ -368,36 +371,25 @@ static void *p11prov_common_gen(struct key_generator *ctx,
...
         ret = p11prov_GenerateRandom(ctx->provctx, sh, id, sizeof(id));
         if (ret != CKR_OK) {
             p11prov_return_session(session);
             return NULL;
         }
-        id_ptr = id;
-        id_len = 16;
+        cka_id.type = CKA_ID;
+        cka_id.pValue = id;
+        cka_id.ulValueLen = 16;
```
[the remark this answers is not in the document]
Comment (simo5) [reply src/keymgmt.c:383]: True, but this is something that can be trivially changed later.

**Pull request merged**